File: ios/LocalVideoViewManager.swift
//
//  LocalVideoViewManager.swift
//

import Foundation

@objc(LocalVideoViewManager)
class LocalVideoViewManager: RCTViewManager {

  override func view() -> UIView! {
    return LocalVideoView.getInstance()
  }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}

File: android/src/main/java/com/omnichannelvoicevideocallingreactnative/LocalVideoViewManager.kt
package com.omnichannelvoicevideocallingreactnative

import android.view.View
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.annotations.ReactProp

class LocalVideoViewManager : SimpleViewManager<View>() {
  override fun getName() = "LocalVideoView"

  override fun createViewInstance(reactContext: ThemedReactContext): View {
    return LocalVideoView.getInstance().createView(reactContext);
  }

  @ReactProp(name = "crop", defaultBoolean = false)
  fun setVideoScalingModeCrop(view: View, crop: Boolean) {
    LocalVideoView.getInstance().setVideoScalingModeCrop(crop);
  }
}
